zoloto/coords.py

```diff
@@ -1,13 +1,84 @@
-from typing import NamedTuple
+from typing import Iterator, NamedTuple, Tuple
 
+from cached_property import cached_property
 from coordinates import spaced_coordinate
+from cv2 import Rodrigues
+from numpy import ndarray
+from pyquaternion import Quaternion
 
 Coordinates = spaced_coordinate("Coordinates", "xy")
 
-Orientation = NamedTuple(
-    "Orientation", [("rot_x", float), ("rot_y", float), ("rot_z", float)]
-)
-
 ThreeDCoordinates = spaced_coordinate("ThreeDCoordinates", "xyz")
 
 Spherical = NamedTuple("Spherical", [("rot_x", float), ("rot_y", float), ("dist", int)])
+
+
+class Orientation:
+    """The orientation of an object in 3-D space."""
+
+    def __init__(self, *rvec) -> None:
```

Comment:
is `rvec` always `rot_x, rot_y, rot_z`? If so would be nice to constrain the arguments (and therefore the types)

Reply:
There was a reason I didn't do this.

Technically they aren't rot_x, rot_y, rot_z in the `rvec`, but I will find the correct terminology and give this another go.

Reply:
0478026